build.gradle

```diff
@@ -291,9 +294,29 @@ task patchExternalModules(type: PatchExternalModules) {
 
 task verifyIsProductionBuildEnvironment {
     doLast {
-        assert javaVersion.java7: "Must use a Java 7 compatible JVM to perform this build. Current JVM is ${jvm}"
+        assert javaVersion.java8:
+            "Must use a Java 8 compatible JVM to perform this build. Current JVM is ${jvm}"
+
+        def javaHomeProperty = { propertyName -> project.findProperty(propertyName) ?: System.getProperty(propertyName) }
```

Comment:
❌ Note that the name of the property (`java7Home`) and the name of the system property (`java7.home`) is different for the compilation property.
I guess we should make sure that the project properties are set at this point.
https://github.com/gradle/gradle/blob/fd3486cde0040289e81903d0ec4971b82e64f0d4/gradle/compile.gradle#L5-L11

Reply:
Addressed in https://github.com/gradle/gradle/pull/3747